src/azure-cli/azure/cli/command_modules/appconfig/_utils.py

```diff
@@ -34,17 +35,49 @@ def construct_connection_string(cmd, config_store_name):
 
 
 def resolve_store_metadata(cmd, config_store_name):
+    resource_group = None
+    endpoint = None
     try:
         config_store_client = cf_configstore(cmd.cli_ctx)
         all_stores = config_store_client.list()
         for store in all_stores:
             if store.name.lower() == config_store_name.lower():
-                # Id has a fixed structure /subscriptions/subscriptionName/resourceGroups/groupName/providers/providerName/configurationStores/storeName"
-                return store.id.split('/')[4], store.endpoint
+                if resource_group is None:
+                    # Id has a fixed structure /subscriptions/subscriptionName/resourceGroups/groupName/providers/providerName/configurationStores/storeName"
+                    resource_group = store.id.split('/')[4]
+                    endpoint = store.endpoint
+                else:
+                    raise ValidationError('Multiple configuration stores found with name {}.'.format(config_store_name))
     except HttpResponseError as ex:
-        raise CLIError("Failed to get the list of App Configuration stores for the current user. Make sure that the account that logged in has sufficient permissions to access the App Configuration store.\n{}".format(str(ex)))
+        raise AzureInternalError("Failed to get the list of App Configuration stores for the current user. Make sure that the account that logged in has sufficient permissions to access the App Configuration store.\n{}".format(str(ex)))
```

Comment:
>AzureInternalError [](http://example.com/codeflow?start=14&length=18)

We shouldn't throw AzureInternalError as that is meant for  5xx server errors only

Reply:
How about AzureResponseError?

Reply:
Sounds good.